pallets/price-feed/src/lib.rs

```diff
@@ -244,10 +244,16 @@ pub mod pallet {
             base: T::AssetId,
             quote: T::AssetId,
         ) -> Result<AssetPricePair<T::AssetId>, DispatchError> {
-            let base_feed_id =
-                Self::get_asset_feed_id(&base).ok_or(Error::<T>::AssetPriceFeedNotFound)?;
-            let quote_feed_id =
-                Self::get_asset_feed_id(&quote).ok_or(Error::<T>::AssetPriceFeedNotFound)?;
+            let (base_feed_id, quote_feed_id) = match (
+                Self::get_asset_feed_id(&base),
+                Self::get_asset_feed_id(&quote),
+            ) {
+                (None, Some(_)) | (Some(_), None) | (None, None) => {
+                    return <InitialPricePairs<T>>::get(&quote)
+                        .ok_or(Error::<T>::AssetPriceFeedNotFound.into())
+                }
+                (Some(b), Some(q)) => (b, q),
```

Comment:
since we have only 2 separate arms here, can we simplify this with `if let (Some(b), Some(q)) = (Self::get_asset_feed_id(&base), Self::get_asset_feed_id(&quote))`?